when looking for latest fastly VCL to clone, use only active VCLs

benjiwheeler opened the pull request:
### Resolves:

Resolves https://github.com/LLK/scratch-www/issues/1342

### Changes:

When comparing fastly vcls to find most recent one, prefer one vcl to another if and only if it is both higher-numbered and active. (Previously, we only considered number.)

### Test Coverage:

Added a test.

colbygk commented on bin/lib/fastly-extended.js:
```diff
@@ -39,7 +39,14 @@ module.exports = function (apiKey, serviceId) {
             }
             var latestVersion = versions.reduce(function (lateVersion, version) {
```
I'm surprised our linter didn't warn/error on this. Don't we prefer (lateVersion, version) => { ... } ? Also, the var statements here ought to be const's. But that is pre-existing, so a new issue and not this PR.

colbygk replied:
Also to avoid confusion, I think the two objects being compared in the reduce() function ought not be named the same as the object being returned via callback at line 52 and defined/getting return from the reduce() at line 40.

colbygk replied:
(sorry, just noticed they're not the same name, just close named)

colbygk commented on test/unit/lib/fastly-extended.test.js:
```diff
@@ -0,0 +1,44 @@
+                {
+                    number: 2,
+                    active: true
+                },
```
I don think this is good test data. Two versions cannot be active at the same time for a single service by definition no?

benjiwheeler replied:
Good point, changing!

benjiwheeler commented:
OK, I revised and (I think) simplified the `latestActiveVersionSoFar()` code, made the tests only include one active version, and added more tests for edge cases. I also changed `latestActiveVersionSoFar()` so that if there is no active version, it returns null.
